modules/garbage-collection.adoc

```diff
@@ -0,0 +1,117 @@
+
+For each type of garbage collection, {productname} provides metrics for the number of rows per table deleted by each garbage collection worker. The following image shows an example of how {productname} monitors garbage collection with the same metrics:
+
+image:garbage-collection-metrics.png[Garbage collection metrics]
+
+=== Measuring storage reclamation
+
+{productname} does not have a way to track how much space is freed up by garbage collection. Currently, the best sign of this is by checking how many blobs have been deleted in the provided metrics.
+
+[NOTE]
+====
+The `UploadedBlob` table in the {productname} metrics tracks the various blobs that are associated with a repository. When a blob is uploaded, it will not be garbage collected before the time designated by the `PUSH_TEMP_TAG_EXPIRATION_SEC` parameter. This is to avoid prematurely deleting blobs that are part of an ongoing push. For example, if garbage collection is set to run often, and a tag is deleted in the span of less than one hour, then it is possible that the associated blobs will not get cleaned up immediately. Instead, and assuming that the time designated by the `PUSH_TEMP_TAG_EXPIRATION_SEC` parameter has passed, the associated blobs will be removed the next time garbage collection runs on that same repository.
+====
+
+== Garbage collection configuration fields
+
+The following configuration fields are available to customize what is garbage collected, and the frequency at which garbage collection occurs:
+
+[cols="4a,2a,2a",options="header"]
+|===
+|Name |Description |Schema
+| **FEATURE_GARBAGE_COLLECTION** | Whether garbage collection is enabled for image tags. Defaults to `true`. |Boolean
+| **FEATURE_NAMESPACE_GARBAGE_COLLECTION** | Whether garbage collection is enabled for namespaces. Defaults to `true`. |Boolean
+| **FEATURE_REPOSITORY_GARBAGE_COLLECTION** | Whether garbage collection is enabled for repositories. Defaults to `true`. |Boolean
+| **GARBAGE_COLLECTION_FREQUENCY** | The frequency, in seconds, at which the garbage collection worker runs. Affects only garbage collection workers. Defaults to 30 seconds. |String
+| **PUSH_TEMP_TAG_EXPIRATION_SEC** | The number of seconds that blobs will not be garbage collected after being uploaded. This feature prevents garbage collection from cleaning up blobs that are not referenced yet, but still used as part of an ongoing push. |String
+| **TAG_EXPIRATION_OPTIONS** | List of valid tag expiration values. |String
+| **DEFAULT_TAG_EXPIRATION** | Tag expiration time for time machine. |String
+
+|===
+
+== Disabling garbage collection
+
+The garbage collection features for image tags, namespaces, and repositories are stored in the `config.yaml`. These features default to `true`.
+
+Disabling garbage collection is not recommended. If disabled, dangling or untagged images, repositories, namespaces, layers, and manifests are not removed. This might increase the downtime of your environment.
```

Comment:
First sentence: "Do not disable garbage collection."
https://www.ibm.com/docs/en/ibm-style?topic=information-claims-recommendations